clients/da-vinci-client/src/main/java/com/linkedin/davinci/kafka/consumer/ActiveActiveStoreIngestionTask.java

```diff
@@ -746,10 +742,9 @@ protected void leaderExecuteTopicSwitch(
         if (topicSwitch.rewindStartTimestamp == REWIND_TIME_DECIDED_BY_SERVER) {
           rewindStartTimestamp = calculateRewindStartTime(partitionConsumptionState);
           LOGGER.info(
-              "{} leader calculated rewindStartTimestamp {} for topic {} partition {}",
+              "{} leader calculated rewindStartTimestamp {} for {}",
```

Comment:
`PubSubTopicPartition` doesn't have a default `toString()` implementation because it is not allowed... If I do this in an interface:

```
  @Override
  default String toString() {
    return "blah";
  }
```

I get the following error: `Default method 'toString' overrides a member of 'java.lang.Object'`

My understanding is that the default method of the interface would never come into play because every `Object` always has a `toString()` function already. If we want to have it there, it would need to be an abstract class rather than an interface.

I do have a `toString()` on `PubSubTopicPartitionImpl`, however.

Reply:
Ok, where the standard is that Impl's of the interface have a toString() method that logs something hopefully human parsable....  I think my only hang up is that is it desirable from an ops standpoint to have that log diverge depending on the particular implementation?  I guess in a typical deployment it's unlikely that you'd need to support two different pubsub technologies concurrently so it'll standardize by virtue of the Impl du jour.... Hrm.  By default I think it'll just print the hexcode value of the Impl if it's not done.  Whats probably more practically going to happen is the next person that tries to write a new Impl will look at an existing Impl and hopefully take note of the toString method.  Maybe just add a comment in there or somewhere that it's desirable to implement it to as to take advantage of sane log messages?

Reply:
I don't think we need more than one implementation of this interface, regardless of which pub sub is in play. I can get rid of the interface entirely and use just the class directly if you're feeling anxious about it.